### PR TITLE
feat: add GD image backend

### PR DESCRIPTION
**Pull request by arxeiss**
Copy of closed #114 

---

My client's web hosting provider don't have support for Imagick, even though they are promising it will be added.

I'm generating QR codes for payment, which is added into email body. And email client's do not have nice support for SVG. So I need old-school GD support.

However, GD has limited sets of functions. I believe, that it is OK, when that Renderer will have also limited sets of features.
So this is not supported:

- Any roundness like DotsModule, RoundnessModule or SimpleCircleEye
- Gradients

And because GD does not support Paths like Imagick or SVG do, I need to generate a bit different paths, so they are not so complex.

---

Sorry for opening new PR, but I had to change branch. I really needed the GD Renderer and in many projects I'm using this in `package.json`. Not to break them by changing commit name, I had to create new branch. But then it is not possible to change source branch in PR.

```json
"repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/arxeiss/BaconQrCode"
        }
    ],
    "require": {
        "php": "^8.2",
        "bacon/bacon-qr-code": "dev-add-gd-backend#650e10121f4d8707ecdc5984b042cae7731e388a",
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,3 +37,21 @@ BaconQrCode comes with multiple back ends for rendering images. Currently includ
 - `ImagickImageBackEnd`: renders raster images using the Imagick library
 - `SvgImageBackEnd`: renders SVG files using XMLWriter
 - `EpsImageBackEnd`: renders EPS files
+
+### GDLib Renderer
+GD library has so many limitations, that GD support is not added as backend, but as separated rendered.
```
Comment (DASPRiD) [inline README.md:42]: `s/rendered/renderer/`

File: README.md [hunk 2]
```diff
@@ -37,3 +37,21 @@ BaconQrCode comes with multiple back ends for rendering images. Currently includ
...
+Use `GDLibRendered` instead of `ImageRenderer`. Here are limitations:
```
Comment (DASPRiD) [inline README.md:43]: `s/Rendered/Renderer/`

File: src/Renderer/GDLibRenderer.php [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+            $this->colors[$colorKey] = $colorId;
+        }
+
+        return $this->colors[$colorKey];
+    }
+
+    private function imageFilledPolygon(array $points, int $color): bool
+    {
+        if (\PHP_VERSION_ID >= 80000) {
```
Comment (DASPRiD) [inline src/Renderer/GDLibRenderer.php:241]: This check should not be required anymore, as we raised the minimum version to 8.1 in the next release.

**Review by DASPRiD — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (DASPRiD) [reply README.md:43]: Oh, and: "**These** are **the** limitations" :)

File: README.md (hunk 1 above)
Comment (arxeiss) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (arxeiss) [reply README.md]: Fixed

File: src/Renderer/GDLibRenderer.php (hunk 3 above)
Comment (arxeiss) [reply src/Renderer/GDLibRenderer.php]: Removed

**Review by DASPRiD — APPROVED** [review]

**Comment by DASPRiD** [comment]
LGTM, thanks for your work! :)

**Comment by vlakoff** [comment]
The `gd` extension has been added to `.github/workflows/ci.yml`, but it should have been added to `.github/workflows/pull-request.yml` as well for consistency.

**Pull request merged**